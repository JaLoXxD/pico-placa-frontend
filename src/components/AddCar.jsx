import { useState } from "react";
import { useFetchApi } from "../hooks/useFetchApi";

export const AddCar = ({ onCheckForm, visible }) => {
	const [carPlate, setCarPlate] = useState("");
	const [color, setColor] = useState("");
	const [model, setModel] = useState("");
	const [chasis, setChasis] = useState("");
	const [year, setYear] = useState("");

	const { newCar, isLoading } = useFetchApi();

	const onCarPlateChange = ({ target }) => {
		setCarPlate(target.value);
	};

	const onColorChange = ({ target }) => {
		setColor(target.value);
	};

	const onModelChange = ({ target }) => {
		setModel(target.value);
	};

	const onChasisChange = ({ target }) => {
		setChasis(target.value);
	};

	const onYearChange = ({ target }) => {
		setYear(target.value);
	};

	

	const onSubmit = async (e) => {
		e.preventDefault();
		const carInfo = {
			placa: carPlate,
			color: color,
			modelo: model,
			chasis: chasis,
			anio: String(year),
		};
		const data = await newCar(carInfo);
		onCheckForm(data);
	};

	return (
		<div onSubmit={onSubmit} className={`formContainer ${!visible && "hideContainer"}`}>
			<form>
				<div className="customInput">
					<label htmlFor="carPlate">Car Plate:</label>
					<input className="test" id="carPlate" type="text" value={carPlate} onChange={onCarPlateChange} />
				</div>
				<div className="customInput">
					<label htmlFor="color">Color:</label>
					<input className="test" id="color" type="text" value={color} onChange={onColorChange} />
				</div>
				<div className="customInput">
					<label htmlFor="model">Model:</label>
					<input className="test" id="model" type="text" value={model} onChange={onModelChange} />
				</div>
				<div className="customInput">
					<label htmlFor="chasis">Chasis:</label>
					<input className="test" id="chasis" type="text" value={chasis} onChange={onChasisChange} />
				</div>
				<div className="customInput">
					<label htmlFor="year">Year:</label>
					<input className="test" id="year" type="number" value={year} onChange={onYearChange} />
				</div>
				<input type="submit" value="Add Car" className="btn btn-success" />
			</form>
		</div>
	);
};
