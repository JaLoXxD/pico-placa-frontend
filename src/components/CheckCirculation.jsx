import { useEffect, useState } from "react";
import { useFetchApi } from "../hooks/useFetchApi";

export const CheckCirculation = ({ onCheckCirculation, visible }) => {
	const now = new Date();
	const formatedDate = new Date(now.getTime() - now.getTimezoneOffset() * 60000).toISOString().substring(0, 19);
	const [carPlate, setCarPlate] = useState("");
	const [currentDate, setCurrentDate] = useState(formatedDate);
	const [circulation, setCirculation] = useState();
	const [cars, setCars] = useState([]);

	const { checkCarPlate, isLoading, allCars } = useFetchApi();

	const fetchCars = async () => {
		const resp = await allCars();
		setCars(resp);
		setCarPlate(resp[0].placa)
	};

	useEffect(() => {
		fetchCars();
	}, []);

	const onCarPlateChange = ({ target }) => {
		setCarPlate(target.value);
		setCurrentDate(new Date(now.getTime() - now.getTimezoneOffset() * 60000).toISOString().substring(0, 19));
	};
	const onDateChange = ({ target }) => {
		setCurrentDate(target.value);
	};
	const onSubmit = async (e) => {
		e.preventDefault();
		const info = await checkCarPlate({ placa: carPlate, date: currentDate });
		setCirculation(info);
		onCheckCirculation(info);
	};

	return (
		<>
			<div onSubmit={onSubmit} className={`formContainer ${!visible && "hideContainer"}`}>
				<form>
					<div className="customInput">
						<label htmlFor="carPlate">Car Plate:</label>
						<select name="car" id="carPlate" value={carPlate} onChange={onCarPlateChange}>
							{cars.map(({ _id, placa }) => {
								return (
									<option key={_id} value={placa}>
										{placa}
									</option>
								);
							})}
						</select>
					</div>
					<div className="customInput">
						<label htmlFor="date">Date:</label>
						<input className="test" id="date" type="datetime-local" value={currentDate} onChange={onDateChange} />
					</div>
					<input type="submit" value="Check" className="btn btn-success" />
				</form>
			</div>
		</>
	);
};
