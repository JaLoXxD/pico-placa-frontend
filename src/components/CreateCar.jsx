export const CreateCar = () => {
	return (
		<>
			<div className="formContainer">
				<form>
					<div className="customInput">
						<label htmlFor="placa">Placa:</label>
						<input className="test" id="placa" type="text" />
					</div>
					<div className="customInput">
						<label htmlFor="fecha">Fecha y Hora:</label>
						<input className="test" id="fecha" type="datetime-local" />
					</div>
					<input type="submit" value="Search" className="btn btn-success" />
				</form>
			</div>
		</>
	);
};
