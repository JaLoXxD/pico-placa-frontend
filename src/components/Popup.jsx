import { useEffect, useRef } from "react";
import ApproveGif from "../assets/images/approve.gif";
import ErrorGif from "../assets/images/error.gif";
export const Popup = ({ title, description, allowed, visible, onClosePopup }) => {
	const popupContainer = useRef();

	useEffect(() => {
		const closePopup = ({ path }) => {
			if (!path.includes(popupContainer.current)) {
				onClosePopup();
			}
		};
		document.body.addEventListener("click", closePopup);
		return () => {
			document.body.removeEventListener("click", closePopup);
		};
	}, [visible]);

	return (
		<div id="popupContainer" role="popupContainer" className={`${visible && "showPopup"}`}>
			<div className="popupContent" ref={popupContainer}>
				<img role="iconImg" src={allowed ? ApproveGif : ErrorGif} alt="" />
				<h2>{title}</h2>
				<div role="descCont">{description}</div>
			</div>
		</div>
	);
};
