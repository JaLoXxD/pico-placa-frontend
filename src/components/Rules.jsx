export const Rules = () => {
	return (
		<div className="rulesCont">
			<p>The pico & placa schedules are from 06:00 to 09:30 and from 16:00 to 21:00.</p>
			<table className="table table-striped table-responsive">
				<thead>
					<tr>
						<th>Day</th>
						<th>Car Plates that cannot circulate</th>
					</tr>
				</thead>
				<tbody>
					<tr>
						<td>Monday</td>
						<td>Car Plates ending in 1 and 2</td>
					</tr>
					<tr>
						<td>Tuesday</td>
						<td>Car Plates ending in 3 and 4</td>
					</tr>
					<tr>
						<td>Wednesday</td>
						<td>Car Plates ending in 5 and 6</td>
					</tr>
					<tr>
						<td>Thursday</td>
						<td>Car Plates ending in 7 and 8</td>
					</tr>
					<tr>
						<td>Friday</td>
						<td>Car Plates ending in 9 and 0</td>
					</tr>
					<tr>
						<td>Saturay</td>
						<td>Free circulation</td>
					</tr>
					<tr>
						<td>Sunday</td>
						<td>Free circulation</td>
					</tr>
				</tbody>
			</table>
		</div>
	);
};
