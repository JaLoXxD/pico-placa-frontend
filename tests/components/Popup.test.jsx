import { render, screen, fireEvent } from "@testing-library/react";
import { Popup } from "../../src/components/Popup";

jest.mock("../../src/assets/images/approve.gif", () => {
	return {
		default: "approve.gif",
	};
});
jest.mock("../../src/assets/images/error.gif", () => {
	return {
		default: "error.gif",
	};
});

jest.mock("../../src/assets/images/approve.gif");
jest.mock("../../src/assets/images/error.gif");

describe("tests on Popup component", () => {
	test("should make match with the snapshot", () => {
		const { container } = render(<Popup />);
		expect(container).toMatchSnapshot();
	});

	test("should have a title in h2 and description in div", () => {
		const title = "Test title";
		const description = "I am the description";
		render(<Popup title={title} description={description} />);
		expect(screen.getByRole("heading", { level: 2 }).innerHTML).toBe(title);
		expect(screen.getByRole("descCont").innerHTML).toBe(description);
	});

	test("should have class showPopup if visible is true", () => {
		render(<Popup visible={true} />);
		expect(screen.getByRole("popupContainer").classList).toContain("showPopup");
	});
});
