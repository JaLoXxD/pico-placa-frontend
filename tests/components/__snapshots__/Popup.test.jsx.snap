// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on Popup component should make match with the snapshot 1`] = `
<div>
  <div
    class="undefined"
    id="popupContainer"
    role="popupContainer"
  >
    <div
      class="popupContent"
    >
      <img
        alt=""
        role="iconImg"
        src="[object Object]"
      />
      <h2 />
      <div
        role="descCont"
      />
    </div>
  </div>
</div>
`;
